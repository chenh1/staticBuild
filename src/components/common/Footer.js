import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';

class Footer extends React.Component{
  render() {
    return (
        <div className="footer">
            <ul className="footer-list">
                <li>
                    <a id="mailLink" className="footer-links">
                        <img src="../assets/icons/email.png" data-src="../assets/icons/email-hover.png" alt="Email"></img>
                    </a>
                </li>
                    <a data-link-type="externalLink" href="https://www.linkedin.com/in/tracy-lin" className="footer-links page-link">
                        <img src="../assets/icons/linkedin.png" data-src="../assets/icons/linkedin-hover.png" alt="LinkedIn"></img>
                    </a>
                <li>
                    <a data-link-type="externalLink" href="https://medium.com/@tracy.lin" className="footer-links page-link">
                        <img src="../assets/icons/medium.png" data-src="../assets/icons/medium-hover.png" alt="Medium"></img>
                    </a>
                </li>
                <li>
                    <a data-link-type="externalLink" href="http://cargocollective.com/tracylin" className="footer-links page-link">
                        <img src="../assets/icons/cargo.png" data-src="../assets/icons/cargo-hover.png" alt="Cargo"></img>
                    </a>
                </li>
            </ul>
        </div>
    );
  }
}

export default Footer;
