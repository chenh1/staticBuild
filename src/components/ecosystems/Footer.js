import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';
import CommonContentApi from '../../api/commonContentServices';

class Footer extends React.Component {
    constructor() {
        super();

        this.state = {};
        this.iconHover = this.iconHover.bind(this);
    }

    componentWillMount() {
        let newState = CommonContentApi.getFooterIcons();
        this.setState(newState);
    }

    iconHover(e) {
        let stateCopy = Object.assign({}, this.state);
        let index = e.currentTarget.getAttribute('data-index');
        let icon = stateCopy.icons[index];
        icon.imageSrc = e.type === "mouseover" ?
            icon.hoverImageSrc : icon.regularImageSrc

        this.setState(stateCopy);
    }

    render() {
        return (
            <div className="footer">
                <ul className="footer-list">
                    {this.state.icons.map((icon, index) =>
                        <li key={"footerLink" + index}>
                            <a onMouseLeave={this.iconHover} 
                                onMouseOver={this.iconHover} 
                                data-index={index} 
                                href={icon.link} 
                                className="footer-links">
                                <img src={icon.imageSrc} alt={icon.alt} />
                            </a>
                        </li>
                    )}
                </ul>
            </div>
        );
    }
}

export default Footer;
