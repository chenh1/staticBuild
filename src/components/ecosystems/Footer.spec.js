import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Footer from './Footer';

describe('<Footer />', () => {
    test('renders Footer', () => {
        const tree = renderer.create(<Footer />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('icon should change image src on mouse over', () => {
        //arrange
        const wrapper = shallow(<Footer />);
        const currentClass = wrapper.find('a[data-index=1]').find('img').classList;
        wrapper.state = {
            icons: [
                {
                    alt: "email",
                    link: "#",
                    imageSrc: '/mockSrc0',
                    regularImageSrc: '/mockSrc0',
                    hoverImageSrc: '/mockSrcHover0'
                },
                {
                    alt: "mockIcon",
                    link: "/mockLink",
                    imageSrc: '/mockSrc1',
                    regularImageSrc: '/mockSrc1',
                    hoverImageSrc: '/mockSrcHover1'
                }
            ]
        };

        //act
        wrapper.find('a[data-index=1]')
            .simulate('mouseover', {currentTarget: {getAttribute: () => 1}});

        //assert
        expect(wrapper.state.icons[1].imageSrc).toEqual('');
    })
});