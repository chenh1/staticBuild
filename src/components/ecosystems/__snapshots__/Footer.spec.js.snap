// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders Footer 1`] = `
<div
  className="footer"
>
  <ul
    className="footer-list"
  >
    <li>
      <a
        className="footer-links"
        data-index={0}
        href="#"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="email"
          src=""
        />
      </a>
    </li>
    <li>
      <a
        className="footer-links"
        data-index={1}
        href="https://www.linkedin.com/in/tracy-lin"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="linkedIn"
          src=""
        />
      </a>
    </li>
    <li>
      <a
        className="footer-links"
        data-index={2}
        href="https://medium.com/@tracy.lin"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="medium"
          src=""
        />
      </a>
    </li>
    <li>
      <a
        className="footer-links"
        data-index={3}
        href="http://cargocollective.com/tracylin"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <img
          alt="cargo"
          src=""
        />
      </a>
    </li>
  </ul>
</div>
`;
